xrdp/xrdp_wm.c

```diff
@@ -1230,6 +1230,29 @@ xrdp_wm_clear_popup(struct xrdp_wm *self)
     return 0;
 }
 
+/*****************************************************************************/
+int
+xrdp_wm_mouse_touch(struct xrdp_wm *self, int gesture, int param)
```

Comment:
Okay, I'm hesitated here because I don't know the best place to put the macros. Should I put them in the 'xrdp_constant.h'?

Reply:
Since `xrdp_wm_mouse_touch()` is a local function only used in this file I'd add them directly above the definition of teh function. Also, the function could be a `static int` rather than an `int`, or have I missed something?

Reply:
I tought it's the same as `xrdp_wm_mouse_click()` and `xrdp_wm_key()`. Make a decleration in 'xrdp.h' or still local method?

Reply:
That's fine too - it's at least consistent with the other functions.